src/nbl/asset/utils/CHLSLCompiler.cpp

```diff
@@ -61,19 +71,71 @@ CHLSLCompiler::~CHLSLCompiler()
 }
 
 
-struct DxcCompilationResult
+static void try_upgrade_hlsl_version(std::vector<std::wstring>& arguments) 
 {
-    ComPtr<IDxcBlobEncoding> errorMessages;
-    ComPtr<IDxcBlob> objectBlob;
-    ComPtr<IDxcResult> compileResult;
+    auto foundShaderStageArgument = std::find(arguments.begin(), arguments.end(), L"-HV");
+    if (foundShaderStageArgument != arguments.end()) {
+        auto foundShaderStageArgumentValueIdx = foundShaderStageArgument - arguments.begin() + 1;
+        std::wstring version = arguments[foundShaderStageArgumentValueIdx];
+        if (version.length() >= 4 && (version[2] < '2' || (version[2] == '2' && version[3] <= '0'))) 
+            arguments[foundShaderStageArgumentValueIdx] = L"2021";
+    }
+}
 
-    std::string GetErrorMessagesString()
+
+static void try_upgrade_shader_stage(std::vector<std::wstring> &arguments) {
+    auto foundShaderStageArgument = std::find(arguments.begin(), arguments.end(), L"-T");
+    if (foundShaderStageArgument != arguments.end()) {
+        auto foundShaderStageArgumentValueIdx = foundShaderStageArgument - arguments.begin() + 1;
+        std::wstring targetProfile = arguments[foundShaderStageArgumentValueIdx];
+        if (targetProfile.length() >= 6) {
+            int argument_version = (targetProfile[3] - '0') * 10 + (targetProfile[5] - '0');
+            if (argument_version < 67)
+            {
+                targetProfile.replace(3, 3, L"6_7");
+                arguments[foundShaderStageArgumentValueIdx] = targetProfile;
+            }
+        }
+    }
+}
+
+
+static void add_required_arguments_if_not_present(std::vector<std::wstring>& arguments, system::logger_opt_ptr &logger) {
+    bool found_arg_flags[CHLSLCompiler::RequiredArgumentCount]{};
+    int argc = arguments.size();
+    for (int i = 0; i < argc; i++)
     {
-        return std::string(reinterpret_cast<char*>(errorMessages->GetBufferPointer()), errorMessages->GetBufferSize());
+        for (int j = 0; j < CHLSLCompiler::RequiredArgumentCount; j++)
+        {
+            if (arguments[i] == CHLSLCompiler::RequiredArguments[j]) {
+                found_arg_flags[j] = true;
+                break;
+            }
+        }
```

Comment:
use `std::find_if` or declare an `static map<std::wstring_view>` initialized from the constexpr in the function